Add AttachmentWideLayoutEnabled to WKWebViewConfiguration

squelart opened the pull request:
#### cb1d57a0fec3835fdb8a2969e0e804652df010dc
<pre>
Add AttachmentWideLayoutEnabled to WKWebViewConfiguration
<a href="https://bugs.webkit.org/show_bug.cgi?id=252768">https://bugs.webkit.org/show_bug.cgi?id=252768</a>
rdar://105797723

Reviewed by Tim Horton.

* Source/WebKit/UIProcess/API/Cocoa/WKWebView.mm:
(-[WKWebView _setupPageConfiguration:]):
* Source/WebKit/UIProcess/API/Cocoa/WKWebViewConfiguration.mm:
(-[WKWebViewConfiguration init]):
(-[WKWebViewConfiguration copyWithZone:]):
(-[WKWebViewConfiguration _attachmentWideLayoutEnabled]):
(-[WKWebViewConfiguration _setAttachmentWideLayoutEnabled:]):
* Source/WebKit/UIProcess/API/Cocoa/WKWebViewConfigurationPrivate.h:
* Tools/DumpRenderTree/TestOptions.cpp:
(WTR::TestOptions::defaults):
* Tools/WebKitTestRunner/TestOptions.h:
(WTR::TestOptions::enableAttachmentWideLayout const):
* Tools/WebKitTestRunner/cocoa/TestControllerCocoa.mm:
(WTR::TestController::platformCreateWebView):

Canonical link: <a href="https://commits.webkit.org/260791@main">https://commits.webkit.org/260791@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/39a67b0330f02a8e1ba4e21d0615b05fae316174

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/109352 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/18427 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/42151 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/87/builds/884 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/118551 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/113235 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/19971 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/9705 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/101586 "Built successfully") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/115108 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/14892 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/98114 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/43075 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/96864 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/29767 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/84827 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/11251 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/31110 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/11924 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/8044 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/17291 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/50713 "Passed tests") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/13551 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/75/builds/4076 "Built successfully and passed tests") | | | | 
<!--EWS-Status-Bubble-End-->

hortont424 commented on Source/WebKit/UIProcess/API/C/WKPreferencesRefPrivate.h:
```diff
@@ -408,6 +408,10 @@ WK_EXPORT bool WKPreferencesGetAllowsPictureInPictureMediaPlayback(WKPreferences
 WK_EXPORT bool WKPreferencesGetAttachmentElementEnabled(WKPreferencesRef preferencesRef);
 
+// Defaults to false.
+WK_EXPORT void WKPreferencesSetAttachmentWideLayoutEnabled(WKPreferencesRef preferencesRef, bool flag);
```
Do we need to extend the C API? It seems like no, so let's not.

hortont424 commented on Source/WebKit/UIProcess/API/Cocoa/WKWebViewConfigurationPrivate.h:
```diff
@@ -80,6 +80,7 @@ typedef NS_ENUM(NSUInteger, _WKContentSecurityPolicyModeForExtension) {
 @property (nonatomic, setter=_setInvisibleAutoplayNotPermitted:) BOOL _invisibleAutoplayNotPermitted WK_API_AVAILABLE(macos(10.12), ios(10.0));
 @property (nonatomic, setter=_setMediaDataLoadsAutomatically:) BOOL _mediaDataLoadsAutomatically WK_API_AVAILABLE(macos(10.12), ios(10.0));
 @property (nonatomic, setter=_setAttachmentElementEnabled:) BOOL _attachmentElementEnabled WK_API_AVAILABLE(macos(10.12), ios(10.0));
+@property (nonatomic, setter=_setAttachmentWideLayoutEnabled:) BOOL _attachmentWideLayoutEnabled WK_API_AVAILABLE(macos(13.0), ios(16.0));
```
WK_MAC_TBA/WK_IOS_TBA

hortont424 commented on Source/WebKitLegacy/mac/WebView/WebPreferences.mm:
```diff
@@ -2564,6 +2564,16 @@ - (void)setAttachmentElementEnabled:(BOOL)flag
     [self _setBoolValue:flag forKey:WebKitAttachmentElementEnabledPreferenceKey];
 }
 
+- (BOOL)attachmentWideLayoutEnabled
```
Revert this whole block

hortont424 commented on Source/WebKitLegacy/mac/WebView/WebPreferencesPrivate.h:
```diff
@@ -232,6 +232,7 @@ extern NSString *WebPreferencesCacheModelChangedInternalNotification WEBKIT_DEPR
 @property (nonatomic) BOOL javaScriptMarkupEnabled;
 @property (nonatomic) BOOL mediaDataLoadsAutomatically;
 @property (nonatomic) BOOL attachmentElementEnabled;
+@property (nonatomic) BOOL attachmentWideLayoutEnabled;
```
Revert this whole block

hortont424 commented on Source/WebKitLegacy/mac/WebView/WebPreferenceKeysPrivate.h:
```diff
@@ -225,6 +225,7 @@
 #define WebKitICECandidateFilteringEnabledPreferenceKey @"WebKitICECandidateFilteringEnabled"
 #define WebKitMediaCaptureRequiresSecureConnectionPreferenceKey @"WebKitMediaCaptureRequiresSecureConnection"
 #define WebKitAttachmentElementEnabledPreferenceKey @"WebKitAttachmentElementEnabled"
+#define WebKitAttachmentWideLayoutEnabledPreferenceKey @"WebKitAttachmentWideLayoutEnabled"
```
Revert this whole block

hortont424 commented on Source/WebKit/UIProcess/API/C/WKPreferencesRefPrivate.h:
```diff
@@ -408,6 +408,10 @@ WK_EXPORT bool WKPreferencesGetAllowsPictureInPictureMediaPlayback(WKPreferences
 WK_EXPORT void WKPreferencesSetAttachmentElementEnabled(WKPreferencesRef preferencesRef, bool flag);
 WK_EXPORT bool WKPreferencesGetAttachmentElementEnabled(WKPreferencesRef preferencesRef);
 
+// Defaults to false.
```
Revert this whole block

hortont424 commented on Source/WebKit/UIProcess/API/C/WKPreferences.cpp:
```diff
@@ -846,6 +846,16 @@ bool WKPreferencesGetAttachmentElementEnabled(WKPreferencesRef preferencesRef)
     return toImpl(preferencesRef)->attachmentElementEnabled();
 }
 
+void WKPreferencesSetAttachmentWideLayoutEnabled(WKPreferencesRef preferencesRef, bool flag)
```
Revert this whole block

squelart committed:
Add AttachmentWideLayoutEnabled to WKWebViewConfiguration
https://bugs.webkit.org/show_bug.cgi?id=252768
rdar://105797723

Reviewed by Tim Horton.

* Source/WebKit/UIProcess/API/Cocoa/WKWebView.mm:
(-[WKWebView _setupPageConfiguration:]):
* Source/WebKit/UIProcess/API/Cocoa/WKWebViewConfiguration.mm:
(-[WKWebViewConfiguration init]):
(-[WKWebViewConfiguration copyWithZone:]):
(-[WKWebViewConfiguration _attachmentWideLayoutEnabled]):
(-[WKWebViewConfiguration _setAttachmentWideLayoutEnabled:]):
* Source/WebKit/UIProcess/API/Cocoa/WKWebViewConfigurationPrivate.h:
* Tools/DumpRenderTree/TestOptions.cpp:
(WTR::TestOptions::defaults):
* Tools/WebKitTestRunner/TestOptions.h:
(WTR::TestOptions::enableAttachmentWideLayout const):
* Tools/WebKitTestRunner/cocoa/TestControllerCocoa.mm:
(WTR::TestController::platformCreateWebView):

Canonical link: https://commits.webkit.org/260791@main
